leadbutt.py

```diff
@@ -173,50 +351,90 @@ def get_metric_statistics(**kwargs):
+                    statistics=metric['Statistics'],
+                    dimensions=metric['Dimensions'],
+                    unit=unit
+                )
+
+                if 'NullIsZero' in options and metric_name in options['NullIsZero']:
+                    results = value_pad_results(
+                        results,
+                        start_time,
+                        end_time,
+                        options['NullIsZero'][metric_name],
+                    )
+
+                output_results(results, this_metric, options)
+                time.sleep(kwargs.get('interval', 0) / 1000.0)
+
+    # get enhanced monitoring if it is enabled
+    if enhanced_monitoring:
+        options = get_options(config_options, None, cli_options)
+        # convert minutes to seconds
         period_local = options['Period'] * 60
         count_local = options['Count']
+        log_group = enhanced_monitoring['LogGroup']
+        # determine formatter
+        if 'Formatter' in enhanced_monitoring:
+            options['Formatter'] = enhanced_monitoring['Formatter']
+            options['ListFormatter'] = options['Formatter']
+        # determine if there is a custom formatter for logs in list form
+        if 'ListFormatter' in enhanced_monitoring:
+            options['ListFormatter'] = enhanced_monitoring['ListFormatter']
+
         # if you have metrics that are available only every 5 minutes, be sure to request only stats
         # that are likely/sure to be up to date, ie ones ending on the previous period increment.
-        end_time = datetime.datetime.utcnow() - datetime.timedelta(
-            seconds=int(time.time()) % period_local
-        )
-        start_time = end_time - datetime.timedelta(
-            seconds=period_local * count_local
-        )
-        # if 'Unit 'is in the config, request only that; else get all units
-        unit = metric.get('Unit')
-        metric_names = metric['MetricName']
-        if not isinstance(metric_names, list):
-            metric_names = [metric_names]
-        for metric_name in metric_names:
-            # we need a copy of the metric dict with the MetricName swapped out
-            this_metric = metric.copy()
-            this_metric['MetricName'] = metric_name
-            results = get_metric_statistics(
-                connection=conn,
-                period=period_local,
+        # convert date to miliseconds
+        end_time = int((datetime.datetime.now() - datetime.timedelta(seconds=int(time.time()) % period_local)).strftime("%s")) * 1000
+        start_time = end_time - (period_local * count_local * 1000)
+        # connect to endpoint
+        logs_conn = boto.logs.connect_to_region(region)
+        # get all streams in log group
+        log_streams = logs_conn.describe_log_streams(log_group_name=log_group)
+        # pluck only stream names out
+        streams = map(lambda x: x['logStreamName'], log_streams['logStreams'])
```

Comment:
It's truth. Always prefer list comprehensions over `map` in python - the Python developers commit to tuning list comprehensions for performance, `map` has no such guarantee.